src/nupic/regions/BacktrackingTMRegion.cpp

```diff
@@ -0,0 +1,1191 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013-2018, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ *
+ * Author: David Keeney, June 2018
+ * ---------------------------------------------------------------------
+ */
+#include <cstring> //strncpy
+#include <fstream>
+#include <iomanip> // setprecision() in stream
+#include <iostream>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <nupic/algorithms/Anomaly.hpp>
```

Comment:
> does BackTM need Anomaly? Sounds as a different scope

Yes, the .py code for backTM does anomaly.
```
  // Set output for use with anomaly classification region if in anomalyMode
  if (args_.anomalyMode) {
    Byte *lrn = tm_->getLearnActiveStateT();
    Size size = args_.numberOfCols * args_.cellsPerColumn;
    getOutput("lrnActiveStateT")->getData() = Array(NTA_BasicType_Byte, lrn, size);

    auto activeColumns = VectorHelpers::binaryToSparse(bottomUpIn.asVector<Real32>());
    Real32 anomalyScore = algorithms::anomaly::computeRawAnomalyScore(
        activeColumns, prevPredictedColumns_);
    getOutput("anomalyScore")->getData() = Array(NTA_BasicType_Real32, &anomalyScore, 1);
  }
```

Now, if I do anomaly as a separate plugin then we would probably not want this in here.  Plugins should avoid calling each other because it injects dependencies.  But I don't know if this is really the same.

So I guess I need to defer to you guys if BackTM should be doing "anomalyScore" and "lrnActiveStateT" as outputs.